models/users/user_pii.sql

```diff
@@ -0,0 +1,88 @@
+{{
+    config(
+        materialized="dictionary",
+        schema=env_var("ASPECTS_EVENT_SINK_DATABASE", "event_sink"),
+        fields=[
+            ("user_id", "Int32"),
+            ("external_user_id", "UUID"),
+            ("external_id_type", "String"),
+            ("username", "String"),
+            ("name", "String"),
+            ("email", "String"),
+            ("meta", "String"),
+            ("courseware", "String"),
+            ("language", "String"),
+            ("location", "String"),
+            ("year_of_birth", "String"),
+            ("gender", "String"),
+            ("level_of_education", "String"),
+            ("mailing_address", "String"),
+            ("city", "String"),
+            ("country", "String"),
+            ("state", "String"),
+            ("goals", "String"),
+            ("bio", "String"),
+            ("profile_image_uploaded_at", "String"),
+            ("phone_number", "String"),
+        ],
+        primary_key="(user_id, external_user_id)",
+        layout="COMPLEX_KEY_SPARSE_HASHED()",
+        lifetime=env_var("ASPECTS_PII_CACHE_LIFETIME", "120"),
+        source_type="clickhouse",
+    )
+}}
+
+with
+    most_recent_user_profile as (
+        select
+            user_id,
+            name,
+            email,
+            meta,
+            courseware,
+            language,
+            location,
+            year_of_birth,
+            gender,
+            level_of_education,
+            mailing_address,
+            city,
+            country,
+            state,
+            goals,
+            bio,
+            profile_image_uploaded_at,
+            phone_number,
+            ROW_NUMBER() over (
+                partition by user_id order by (id, time_last_dumped) DESC
+            ) as rn
+        from {{ source("event_sink", "user_profile") }}
+    )
+select
+    mrup.user_id as user_id,
+    external_user_id,
+    external_id_type,
+    username,
+    name,
+    email,
+    meta,
+    courseware,
+    language,
+    location,
+    year_of_birth,
+    gender,
+    level_of_education,
+    mailing_address,
+    city,
+    country,
+    state,
+    goals,
+    bio,
+    profile_image_uploaded_at,
+    phone_number
+from {{ source("event_sink", "external_id") }} ex
+left outer join
```

Comment:
This was changed from a `right outer join` to a `left outer join` so that if a user doesn't have external_id we don't count it as the `external id` is needed to match it against events.